crates/primitives/src/batch.rs

```diff
@@ -345,6 +342,119 @@ impl BatchInfo {
     }
 }
 
+/// Contains transition information in a batch checkpoint, verified by the proof
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Eq,
+    PartialEq,
+    Arbitrary,
+    BorshDeserialize,
+    BorshSerialize,
+    Deserialize,
+    Serialize,
+)]
+pub struct BatchTransition {
+    /// Transition commitment for `Chainstate`
+    pub chainstate_transition: ChainstateRootTransition,
+
+    /// Transition commitment of `TxFilterConfig`
+    pub tx_filters_transition: TxFilterConfigTransition,
+}
+
+/// Contains Chainstate root transition information within a batch, verified by the associated
+/// proof.
+///
+/// This struct represents a concise summary of the Chainstate transition by capturing only the
+/// state roots before and after the execution of a batch of blocks. It serves as an efficient means
+/// to verify state changes without storing the entire Chainstate.
+///
+/// # Example
+///
+/// Given a batch execution transitioning from block `M` to block `N`:
+/// - `pre_state_root` represents the Chainstate root immediately **before** executing block `M`.
+///   i.e. immediately after executing block `M-1`
+/// - `post_state_root` represents the Chainstate root immediately **after** executing block `N`.
```

Comment:
Shouldn't it be `after executing block M`? because otherwise it looks like the following:

`some state ---(apply block M-1)---> pre-state ---(apply block M)---> ?? ---(apply block N) ---> post-state`

Actually, would it make sense  to write "transitioning from block M to block N" as "applying block M"?

Reply:
Got it!